Bottom: 9442c00ce9558ad8c0019a29b786dbd52d3305e2
Top:    563b785042d462b3f63f4267eeee2996ff9ea6b1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-27 18:13:02 +0100

Refresh of silly-space-navigator-interfac

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 0c9019d..6708e42 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -119,7 +119,7 @@ ghid_zoom_view_abs (Coord center_x, Coord center_y, double new_zoom)
   ghid_set_status_line_label ();
 }
 
-static void
+void
 ghid_zoom_view_rel (Coord center_x, Coord center_y, double factor)
 {
   ghid_zoom_view_abs (center_x, center_y, gport->view.coord_per_px * factor);
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index e75ee85..d73ed17 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -732,12 +732,11 @@ ghid_port_window_mouse_scroll_cb (GtkWidget * widget,
   return TRUE;
 }
 
-void zoom_by (double factor, int x, int y); /* XXX: TEMPORARY HACK */
-
 void ndof_pan_cb (int dx, int dy, int dz, gpointer data)
 {
   if (dx || dy)
-    ghid_pan_view_rel (-gport->zoom * 5 * dx, -gport->zoom * 5 * dy);
+    ghid_pan_view_rel (-gport->view.coord_per_px * 5 * dx,
+                       -gport->view.coord_per_px * 5 * dy);
   if (dz)
     ghid_zoom_view_rel (gport->pcb_x, gport->pcb_y, 1.0 - (dz / 100.0));
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 9295964..1a4f5d4 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -516,6 +516,7 @@ void ghid_lead_user_to_location (Coord x, Coord y);
 void ghid_cancel_lead_user (void);
 
 /* gtkhid-main.c */
+void ghid_zoom_view_rel (Coord center_x, Coord center_y, double factor);
 void ghid_pan_view_rel (Coord dx, Coord dy);
 void ghid_get_coords (const char *msg, Coord *x, Coord *y);
 gint PCBChanged (int argc, char **argv, Coord x, Coord y);
